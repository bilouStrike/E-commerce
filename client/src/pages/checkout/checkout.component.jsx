import React from 'react';

import './checkout.styles.scss';
import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors';
import { connect } from 'react-redux';
import CheckoutItem from '../../components/checkout-item/checkout-item.component';
import StripeCheckoutButton from '../../components/stripe-button/stripe-button.component';

const CheckoutPage = ( {cartItems, total} ) => (
  <div className='checkout-page'>
    <div className='checkout-header'>
      <div className='header-block'>
        <span>Product</span>
      </div>
      <div className='header-block'>
        <span>Description</span>
      </div>
      <div className='header-block'>
        <span>Quantity</span>
      </div>
      <div className='header-block'>
        <span>Price</span>
      </div>
      <div className='header-block'>
        <span>Remove</span>
      </div>
    </div>
     { cartItems.map(cartItem => (
        <CheckoutItem key={cartItem.id} cartItem={cartItem} />
       ))
     }
    <div className='total'>TOTAL: ${total}</div>
    <StripeCheckoutButton price ={total} />
  </div>
);


const mapStateToProps = ( state ) => ({
   cartItems: selectCartItems(state),
   total: selectCartTotal(state)
})

export default connect(mapStateToProps)(CheckoutPage);