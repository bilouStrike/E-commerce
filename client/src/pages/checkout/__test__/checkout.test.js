import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { CheckoutPage } from '../checkout.component';

it('expect to render checkout component', () => {
    const mockItems = [{
        imageUrl: 'url',
        price: '250',
        name: 'name1',
        quantity: 2
    }];
    const mockTotal = 500;
    expect(toJson(shallow(<CheckoutPage cartItems={mockItems} total={mockTotal}/>))).toMatchSnapshot();
});