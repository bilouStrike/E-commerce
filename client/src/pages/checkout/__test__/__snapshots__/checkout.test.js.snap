// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render checkout component 1`] = `
<div
  className="checkout-page"
>
  <div
    className="checkout-header"
  >
    <div
      className="header-block"
    >
      <span>
        Product
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Description
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Quantity
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Price
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Remove
      </span>
    </div>
  </div>
  <Connect(CheckoutItem)
    cartItem={
      Object {
        "imageUrl": "url",
        "name": "name1",
        "price": "250",
        "quantity": 2,
      }
    }
  />
  <div
    className="total"
  >
    TOTAL: $
    500
  </div>
  <StripeCheckoutButton
    price={500}
  />
</div>
`;
